### PR TITLE
Gh 3237 check onchain bef reestablish di

### PR DESCRIPTION
**Pull request by velzevur**
Finishes #3255

### CODE REVIEW

**Comment by velzevur** [comment]
I squashed commits and rebased on top of master. Fixed conflicts and rebase artifacts.

**Comment by velzevur** [comment]
I ran out of time for this one, do you support me continuing working on it?

File: apps/aechannel/test/aesc_fsm_SUITE.erl [hunk 1]
```diff
@@ -5181,3 +5252,570 @@ auth_contract_props(btc_auth) ->
...
+    % Verify leave and FSM timeout
+    {ok, #{info := SignedTx}} = await_leave(I, ?TIMEOUT, Debug),
+    {ok, #{info := SignedTx}} = await_leave(R, ?TIMEOUT, Debug),
+    {ok, _} = receive_info(I, fun died_normal/1, Debug),
+    {ok, _} = receive_from_fsm(info, R, fun(#{info := peer_disconnected}) -> ok end,
+                               ?TIMEOUT, Debug),
+    {ok, #{info := {log, _ILog}}} = receive_log(I, Debug),
+    assert_empty_msgq(Debug),
+    ?LOG(Debug, "queue is empty", []),
+    %% post a close solo while the other party is off
```
Comment (uwiger) [inline apps/aechannel/test/aesc_fsm_SUITE.erl:5302]: Up until this point, the code seems to be an excellent candidate for refactoring out a common preamble.

File: apps/aechannel/src/aesc_fsm.erl [hunk 2]
```diff
@@ -3410,15 +3442,21 @@ process_update_error({off_chain_update_error, Reason}, From, D) ->
...
-                                       check_closing_event_(Info, D)
-                                   ?CATCH_LOG(E)
-                                       error(E)
-                                   end
-                           end).
+check_closing_event(St, #{tx := SignedTx} = Info, D) ->
+    {Type, _Tx} = aetx:specialize_type(aetx_sign:innermost_tx(SignedTx)),
+    case Type =:= channel_force_progress_tx of
+        true ->
+            process_incoming_forced_progress(St, Info, D);
```
[the remark this answers is not in the document]
Comment (velzevur) [reply apps/aechannel/src/aesc_fsm.erl:3449]: I think a general refactoring shall be done in FSM that introduces all `async_dirty` in the right places. This one in particular might not be worth it (it has a single call to fetch the club and a call to produce proper transaction context).

File: apps/aechannel/src/aesc_fsm.erl [hunk 3]
```diff
@@ -5428,3 +5517,116 @@ does_force_progress_interrupt(FSMState) ->
...
+                               #data{ role = Role
+                                    , on_chain_id = ChannelId
+                                    , last_channel_change = TxHash } = Data) ->
+    {ok, TxHashes} = aesc_chain_watcher:get_txs_since_tx(ChannelId, TxHash),
+    Data1 = Data#data{ state = State0
+                     , op    = ?NO_OP},
+    {Res, UpdatedData} = apply_non_malicious_txs(TxHashes, Data1),
+    case Res of
+        {malicious, MaliciousTx} ->
+            {ok, Channel} = aec_chain:get_channel(ChannelId),
```
[the remark this answers is not in the document]
Comment (velzevur) [reply apps/aechannel/src/aesc_fsm.erl:5532]: Same comment as above.

File: apps/aechannel/test/aesc_fsm_SUITE.erl [hunk 4]
```diff
@@ -2873,6 +2927,7 @@ match_info(Info, Match) ->
 
 await_open_report(#{fsm := Fsm} = R, Timeout, Debug) ->
     receive {aesc_fsm, Fsm, #{type := report, tag := info, info := open} = Msg} ->
+                ct:log("AAA ~p", [Msg]),
```
[the remark this answers is not in the document]
Comment (velzevur) [reply apps/aechannel/test/aesc_fsm_SUITE.erl]: It stands for "debug, to be removed later on"

**Pull request merged**